src/docker/composeappengine.cc

```diff
@@ -351,6 +360,13 @@ bool ComposeAppEngine::checkAvailableStorageSpace(const boost::filesystem::path&
   return true;
 }
 
+void ComposeAppEngine::verifyAppArchive(const App& app, const std::string& archive_file_name) {
+  if (!cmd_streaming("tar -tf " + archive_file_name + " " + ComposeFile, app) &&
+      !cmd_streaming("tar -tf " + archive_file_name + " ./" + ComposeFile, app)) {
```

Comment:
are both checks really required? seems like the `./` check will never do anything?

Reply:
files can be prepended with `./` prefix in the app archive, both options are valid so I am checking both. The first check expects the compose file without the prefix, which is our default case, so under normal circumstances, just one condition is executed.